pkg/objectfile/pool.go

```diff
@@ -163,6 +163,18 @@ func (p *Pool) Open(path string) (*ObjectFile, error) {
 		}
 		return nil, fmt.Errorf("error opening %s: %w", path, err)
 	}
+
+	// This a fast path to extract the buildID from the ELF header.
```

Comment:
I'm not sure if this will be statically meaningful.

Reply:
I assume you mean statistically?

If so, what made us arrive at this strategy? Why is caching not a viable option?

Reply:
Yes, it's a typo. 

This was a path that we used to take. I have now combined it with what we have recently put in place. 
So maybe the best way to forward is to add a metric for this and see if it's something we observe in the wild.
Or remove it.

The best candidate for this path is dynamically linked shared libraries (like libc) or running the same executable on the host multiple time.

What do you think?

Reply:
I think I would prefer to just see the “slow” option used everywhere and instead use caching for perf improvements.

Reply:
Just to clarify, by slow option, just remove this path and try to open the whole ELF file?